specification/keyvault/data-plane/typespec/models.tsp

```diff
@@ -0,0 +1,956 @@
+   */
+  CustomizedRecoverable: "CustomizedRecoverable",
+
+  /**
+   * Denotes a vault and subscription state in which deletion is recoverable, immediate and permanent deletion (i.e. purge) is not permitted, and in which the subscription itself cannot be permanently canceled when 7<= SoftDeleteRetentionInDays < 90. This level guarantees the recoverability of the deleted entity during the retention interval, and also reflects the fact that the subscription itself cannot be cancelled.
+   */
+  CustomizedRecoverable_ProtectedSubscription: "CustomizedRecoverable+ProtectedSubscription",
+}
+
+/**
+ * The type of key pair to be used for the certificate.
+ */
+union JsonWebKeyType {
+  string,
+  EC: "EC",
+  EC_HSM: "EC-HSM",
+  RSA: "RSA",
+  RSA_HSM: "RSA-HSM",
+  oct: "oct",
+  oct_HSM: "oct-HSM",
+}
+
+/**
+ * Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+ */
+union JsonWebKeyCurveName {
+  string,
+  P_256: "P-256",
+  P_384: "P-384",
+  P_521: "P-521",
+  P_256K: "P-256K",
+}
+
+union KeyUsageType {
+  string,
+  digitalSignature: "digitalSignature",
+  nonRepudiation: "nonRepudiation",
+  keyEncipherment: "keyEncipherment",
+  dataEncipherment: "dataEncipherment",
+  keyAgreement: "keyAgreement",
+  keyCertSign: "keyCertSign",
+  cRLSign: "cRLSign",
+  encipherOnly: "encipherOnly",
+  decipherOnly: "decipherOnly",
+}
+
+/**
+ * The certificate list result.
+ */
+model CertificateListResult is Azure.Core.Page<CertificateItem>;
+
+/**
+ * The certificate item containing certificate metadata.
+ */
+model CertificateItem {
+  /**
+   * Certificate identifier.
+   */
+  id?: string;
+
+  /**
+   * The certificate management attributes.
+   */
+  attributes?: CertificateAttributes;
+
+  /**
+   * Application specific metadata in the form of key-value pairs.
+   */
+  tags?: Record<string>;
+
+  /**
+   * Thumbprint of the certificate.
+   */
+  @encodedName("application/json", "x5t")
+  @encode("base64url")
+  x509Thumbprint?: bytes;
+}
+
+/**
+ * The certificate management attributes.
+ */
+model CertificateAttributes extends Attributes {
+  /**
+   * softDelete data retention days. Value should be >=7 and <=90 when softDelete enabled, otherwise 0.
+   */
+  @visibility("read")
+  recoverableDays?: int32;
+
+  /**
+   * Reflects the deletion recovery level currently in effect for certificates in the current vault. If it contains 'Purgeable', the certificate can be permanently deleted by a privileged user; otherwise, only the system can purge the certificate, at the end of the retention interval.
+   */
+  @visibility("read")
+  recoveryLevel?: DeletionRecoveryLevel;
+}
+
+/**
+ * The object attributes managed by the KeyVault service.
+ */
+model Attributes {
+  /**
+   * Determines whether the object is enabled.
+   */
+  enabled?: boolean;
+
+  /**
+   * Not before date in UTC.
+   */
+  @encodedName("application/json", "nbf")
+  notBefore?: plainTime;
```

Comment:
Is this really just a time with no date?

Reply:
Sorry for the delayed response, I was on vacation. I'll double-check this and update accordingly.

Reply:
Confirmed from recordings that this should be a UTC timestamp; I'll update this to align with other UTC timestamp values